docs/concepts/performance.md

```diff
@@ -0,0 +1,182 @@
+
+### Performance factors
+
+Theoretically, Azure Web PubSub Service capacity is limited by computation resources: CPU, memory, and network. For example, more connections to Azure Web PubSub Service cause the service to use more memory. For larger message traffic (for example, every message is larger than 2,048 bytes), Azure Web PubSub Service needs to spend more CPU cycles to process traffic.
+
+The message routing cost also limits performance. Azure Web PubSub Service plays a role as a message router, which routes the message among a set of clients. A different scenario or API requires a different routing policy. 
+
+For **echo**, the client sends a message to itself, and the routing destination is also itself. This pattern has the lowest routing cost. But for **broadcast**, **send to group**, and **send to connection**, Azure Web PubSub Service needs to look up the target connections through the internal distributed data structure. This extra processing uses more CPU, memory, and network bandwidth. As a result, performance is slower.
+
+In summary, the following factors affect the inbound and outbound capacity:
+
+-   SKU tier (CPU/memory)
+
+-   Number of connections
+
+-   Message size
+
+-   Message send rate
+
+- Use-case scenario (routing cost)
+
+
+### Finding a proper SKU
+
+How can you evaluate the inbound/outbound capacity or find which tier is suitable for a specific use case?
+
+Every tier has its own maximum inbound bandwidth and outbound bandwidth. A smooth user experience is not guaranteed after the inbound or outbound traffic exceeds the limit. 
+
+```
+  inboundBandwidth = inboundConnections * messageSize / sendInterval
+  outboundBandwidth = outboundConnections * messageSize / sendInterval
+```
+
+- *inboundConnections*: The number of connections sending the message.
+- *outboundConnections*: The number of connections receiving the message.
+- *messageSize*: The size of a single message (average value). A small message that's less than 1,024 bytes has a performance impact that's similar to a 1,024-byte message.
+- *sendInterval*: The time of sending one message. Typically it's 1 second per message, which means sending one message every second. A smaller interval means sending more message in a time period. For example, 0.5 seconds per message means sending two messages every second.
+- *Connections*: The committed maximum threshold for Azure Web PubSub Service for every tier. If the connection number is increased further, it will suffer from connection throttling.
+
+Assume that the upstream is powerful enough and is not the performance bottleneck. Then, check the maximum inbound and outbound bandwidth for every tier.
```

Comment:
> Then, check the maximum inbound and outbound bandwidth for every tier.

What does this mean? Is it to "use the maximum inbound and outbound bandwidth to choose the proper SKU"?